lib/node_modules/@stdlib/math/base/special/pow/src/main.c

```diff
@@ -0,0 +1,725 @@
+
+    if ( ahx <= HIGH_MAX_NEAR_UNITY ) {
+        if ( y < 0 ) {
+			// Signal overflow...
+            return HUGE * HUGE;
+        }
+		// Signal underflow...
+        return TINY * TINY;
+    }
+	// `x` has a biased exponent greater than or equal to `0`...
+
+    if ( y > 0 ) {
+		// Signal overflow...
+        return HUGE * HUGE;
+    }
+	// Signal underflow...
+	return TINY * TINY;
+}
+
+/**
+* Evaluates the exponential function when \\(|x| = 0\\).
+*
+* @param x    input value
+* @param y    input value
+* @return	  output value
+*
+* @example
+* double out = x_is_zero( 0.0, 2 );
+* // returns 0.0
+*
+* @example
+* double out = x_is_zero( 0.0, -9 );
+* // returns Infinity
+*/
+double x_is_zero( const double x, const double y ) {
+    if ( y == STDLIB_CONSTANT_FLOAT64_NINF ) {
+        return STDLIB_CONSTANT_FLOAT64_PINF;
+    }
+	if ( y == STDLIB_CONSTANT_FLOAT64_PINF ) {
+		return 0.0;
+	}
+	if ( y > 0.0 ) {
+		if ( stdlib_base_is_odd( y ) ) {
+			return x; // handles +-0
+		}
+		return 0.0;
+	}
+	// y < 0.0
+	if ( stdlib_base_is_odd( y ) ) {
+		return stdlib_base_copysign( STDLIB_CONSTANT_FLOAT64_PINF, x ); // handles +-0
+	}
+	return STDLIB_CONSTANT_FLOAT64_PINF;
+}
+
+/**
+* Computes \\(2^{\mathrm{hp} + \mathrm{lp}\\).
+*
+* @param j    high word of hp + lp
+* @param hp    first power summand
+* @param lp    second power summand
+* @return	  function value
+*
+* @example
+* double out = pow2( 1065961648, -0.3398475646972656, -0.000002438187359100815 );
+* // returns ~0.79
+*/
+double pow2( const uint32_t j, const double hp, const double lp ) {
+	int32_t tmp;
+	double t1;
+	double t;
+	double r;
+	double u;
+	double v;
+	double w;
+	double z;
+	int32_t n;
+	uint32_t i;
+	int32_t k;
+	int32_t jc;
+	uint32_t jcc;
+	double hpc;
+
+	hpc = hp;
+	jc = j;
+	jcc = j;
+	i = ( j & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK );
+	k = ( ( i >> HIGH_NUM_SIGNIFICAND_BITS ) - STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS );
+	n = 0;
+
+	// `|z| > 0.5`, set `n = z+0.5`
+    if ( i > HIGH_BIASED_EXP_NEG_1 ) {
+		n = ( j + ( HIGH_MIN_NORMAL_EXP >> ( k + 1 ) ) ) >> 0; // asm type annotation
+		k = ( ( ( n & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK ) >> HIGH_NUM_SIGNIFICAND_BITS ) - STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS ); // new k for n
+		tmp = ( ( n & !( HIGH_SIGNIFICAND_MASK >> k ) ) ) >> 0; // asm type annotation
+		t = 0.0;
+		stdlib_base_float64_set_high_word( tmp, &t );
+		n = ( ( ( n & HIGH_SIGNIFICAND_MASK ) | HIGH_MIN_NORMAL_EXP ) >> ( HIGH_NUM_SIGNIFICAND_BITS - k ) ) >> 0; // eslint-disable-line max-len
+		if ( jc < 0 ) {
+			n = -n;
+		}
+		hpc -= t;
+    }
+	t = lp + hpc;
+	stdlib_base_float64_set_low_word( 0, &t );
+	u = t * LN2_HI;
+	v = ( ( lp - ( t - hpc ) ) * STDLIB_CONSTANT_FLOAT64_LN2 ) + ( t * LN2_LO );
+	z = u + v;
+	w = v - ( z - u );
+	t = z * z;
+	t1 = z - ( t * polyval_p( t ) );
+	r = ( ( z * t1 ) / ( t1 - 2.0 ) ) - ( w + ( z * w ) );
+	z = 1.0 - ( r - z );
+	stdlib_base_float64_get_high_word( z, &jcc );
+	jc = jcc|0;
```

Comment:
```suggestion
	jc = (int32_t)jcc;
```

Perform an explicit cast.